### PR TITLE
Removing calls to known

### PR DESCRIPTION
**Pull request by pffrandsen**
A lot (if not all) calls to known is actually just a way to bootstrap algorithms with a basic step. Introdcued these basic steps, hence improving performance ever so little.

As far as I can tell, one could reasonably deprecate known.

### CODE REVIEW

File: core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java [hunk 1]
```diff
@@ -46,12 +46,11 @@ public ExponentiationOpenExponent(Computation<SInt> x, BigInteger e) {
   @Override
   public Computation<SInt> build(SequentialNumericBuilder builder) {
     if (exponent.equals(BigInteger.ZERO)) {
-      return builder.numeric().known(BigInteger.valueOf(1));
+      throw new RuntimeException("Now why would you exponentiate a SInt with 0?");
```
Comment (GuutBoy) [inline core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java]: You may want to exponentiate with by a value not known at compile time which could be zero? Of course you could test for that outside of this class and mark it as only compatible with strictly positive exponents. However, in that case this should be an "illegal argument" exception.

Comment (GuutBoy) [inline core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java]: Additionally, the exception should probably be thrown for any value of the exponent that is not strictly positive.

Comment (pffrandsen) [inline core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java]: Agree, branch updated.

Given that this is the open exponentiate, you will now - however maybe not runtime.

Comment (GuutBoy) [inline core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java]: Possibly the error message could be less judgemental as well?

File: core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java [hunk 2]
```diff
@@ -41,13 +41,13 @@
   public ExponentiationOpenExponent(Computation<SInt> x, BigInteger e) {
     this.base = x;
     this.exponent = e;
+    if (exponent.compareTo(BigInteger.ZERO) <= 0) {
+      throw new IllegalArgumentException("Now why would you exponentiate a SInt with 0?");
```
Comment (GuutBoy) [inline core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java]: "... 0 or less"

File: core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java [hunk 3]
```diff
@@ -42,7 +42,8 @@ public ExponentiationOpenExponent(Computation<SInt> x, BigInteger e) {
     this.base = x;
     this.exponent = e;
     if (exponent.compareTo(BigInteger.ZERO) <= 0) {
-      throw new IllegalArgumentException("Now why would you exponentiate a SInt with 0?");
+      throw new IllegalArgumentException(
+          "Now why would you exponentiate a SInt with somthing equal to or less than 0?");
```
Comment (GuutBoy) [inline core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java]: "something"

File: core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java [hunk 4]
```diff
@@ -43,7 +43,7 @@ public ExponentiationOpenExponent(Computation<SInt> x, BigInteger e) {
     this.exponent = e;
     if (exponent.compareTo(BigInteger.ZERO) <= 0) {
       throw new IllegalArgumentException(
-          "Now why would you exponentiate a SInt with somthing equal to or less than 0?");
+          "This computation does not support exponentiate a SInt with somthing equal to or less than 0?");
```
Comment (GuutBoy) [inline core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java]: something

Comment (pffrandsen) [inline core/src/main/java/dk/alexandra/fresco/lib/math/integer/exp/ExponentiationOpenExponent.java]: Changed the wording

**Pull request merged**